Java/CoinChange.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Dynamic Programming ALgorithm for Coin Change Problem.
+ * CoinChangeI:
+ * Given a value N, if we want to make change for N cents, and we have infinite supply of 
+ * each of C = { C1, C2, .. , Cm} valued coins, how many ways can we make the change? 
+ * The order of coins doesn’t matter.
+ * 
+ * Time complexity of this function: O(mn)
+ * Space Complexity of this function: O(n) 
```

Comment:
Use capital letters to describe complexity. e.g.: O(N)

Reply:
Sure. I'll change that and submit the code again.